pkg/util/ssh/socks5_proxy.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Tencent is pleased to support the open source community by making TKEStack
+ * available.
+ *
+ * Copyright (C) 2012-2022 Tencent. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package ssh
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"golang.org/x/net/proxy"
+	"tkestack.io/tke/pkg/util/log"
+)
+
+type SOCKS5 struct {
+	Host        string
+	Port        int
+	DialTimeOut time.Duration
+}
+
+func (sk SOCKS5) ProxyConn(targetAddr string) (net.Conn, func(), error) {
```

Comment:
sk SOCKS5 -> sk *SOCKS5 ?

Reply:
We don't need to change sk's content, we just read it's content, so no pointer is better.

Reply:
since we just read the sk's content, not write, to prevent someone changing its content, no pointer is better.